fs/indexed-pkg-importer/src/index.ts

```diff
@@ -138,15 +139,18 @@ async function hardlinkPkg (
 
 async function linkOrCopy (existingPath: string, newPath: string) {
   try {
-    await fs.link(existingPath, newPath)
+    // If platform is Darwin, use copy instead of link
+    if (platform() === 'darwin') return copyFileSync(existingPath, newPath, constants.COPYFILE_FICLONE_FORCE)
```

Comment:
We already use cloning by default, when it works. Unfortunately, there's an issue in Node.js and cloning doesn't currently work on macOS. See the related issue: https://github.com/pnpm/pnpm/issues/5001